### PR TITLE
Hr/tm 273 view matched criteria of application

### PR DESCRIPTION
**Pull request by hasanrq**

### CODE REVIEW

**Comment by ConorTill** [comment]
I think we're missing the green ticks that are mentioned in the ticket.

File: src/SFA.DAS.LevyTransferMatching.Web/Models/Pledges/ApplicationViewModel.cs [hunk 1]
```diff
@@ -44,6 +44,41 @@ public class ApplicationViewModel : ApplicationPostRequest
...
+            if (this.IsJobRoleMatch) matches++;
+            if (this.IsLevelMatch) matches++;
+            switch (matches)
+            {
+                case 0: case 1: return "pink";
+                case 2: case 3: return "yellow";
+                case 4: return "turquoise";
+                default: return "pink";
+            }
+        }
```
[the remark this answers is not in the document]
Comment (hasanrq) [reply src/SFA.DAS.LevyTransferMatching.Web/Models/Pledges/ApplicationViewModel.cs]: Extension method is created as advised

**Comment by hasanrq** [comment]
> 

Green ticks are implemented, sorry overlooked last time.

File: src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsForDownloadResponse.cs [hunk 2]
```diff
@@ -23,10 +23,11 @@ public class Application
             public string LastName { get; set; }
             public IEnumerable<string> EmailAddresses { get; set; }
             public string BusinessWebsite { get; set; }
-            public bool IsLocationMatch { get; set; }
-            public bool IsSectorMatch { get; set; }
-            public bool IsJobRoleMatch { get; set; }
-            public bool IsLevelMatch { get; set; }
+            public bool MatchSector { get; set; }
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsForDownloadResponse.cs:26]: See comments in APIM layer about changing parts of the service that don't relate to your ticket. Apart from creating potential merge conflicts, the tester will not anticipate this download having changed.

File: src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsResponse.cs [hunk 3]
```diff
@@ -31,10 +31,11 @@ public class Application
             public DateTime CreatedOn { get; set; }
             public ApplicationStatus Status { get; set; }
             public bool IsNamePublic { get; set; }
-            public bool IsLocationMatch { get; set; }
-            public bool IsSectorMatch { get; set; }
-            public bool IsJobRoleMatch { get; set; }
-            public bool IsLevelMatch { get; set; }
+            public bool MatchSector { get; set; }
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsResponse.cs:34]: See previous comment

File: src/SFA.DAS.LevyTransferMatching.Web/Extensions/ApplicationViewModelExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using SFA.DAS.LevyTransferMatching.Web.Models.Pledges;
+
+namespace SFA.DAS.LevyTransferMatching.Web.Extensions
+{
+    public static class ApplicationViewModelExtensions
+    {
+        public static string GetMatchPercentageCss(this ApplicationViewModel viewModel)
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Web/Extensions/ApplicationViewModelExtensions.cs:7]: Does this need to be an extension method? We could just implement this on the viewmodel itself couldn't we?

File: src/SFA.DAS.LevyTransferMatching.Web/Helpers/CriteriaListTagHelper.cs [hunk 5]
```diff
@@ -17,41 +17,27 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
         {
             var content = new StringBuilder();
             var matchesList = new StringBuilder();
-            int matches = 0;
-            if (Application.IsLocationMatch)
+
+            if (Application.MatchLocation)
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Web/Helpers/CriteriaListTagHelper.cs:21]: See previous comments about changing this

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsForDownloadResponse.cs (hunk 2 above)
Comment (hasanrq) [reply src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsForDownloadResponse.cs]: suggestions implemented with as little code changes as needed by this ticket

File: src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsResponse.cs (hunk 3 above)
Comment (hasanrq) [reply src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/PledgeService/Types/GetApplicationsResponse.cs]: suggestions implemented with as little code changes as needed by this ticket

File: src/SFA.DAS.LevyTransferMatching.Web/Helpers/CriteriaListTagHelper.cs (hunk 5 above)
Comment (hasanrq) [reply src/SFA.DAS.LevyTransferMatching.Web/Helpers/CriteriaListTagHelper.cs]: suggestions implemented with as little code changes as needed by this ticket

File: src/SFA.DAS.LevyTransferMatching.Web/Extensions/ApplicationViewModelExtensions.cs (hunk 4 above)
Comment (hasanrq) [reply src/SFA.DAS.LevyTransferMatching.Web/Extensions/ApplicationViewModelExtensions.cs:7]: yes we can and I initially implemented as such. It was requested in first tech review by Conor to be moved out of there as Extension method. I have moved it in view model again now.

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**